### DISCUSSION TITLE
MCP2515 CanSniffer issues, cannot use CanSend.

### DISCUSSION THREAD

**Discussion by AhtiL** [arendst/Tasmota#18287] [Q&A]
Hey, as per title I've tried using the xsns_87_cansniffer module. I have the MCP2515 properly wired and console confirms it by saying the Sniffer has been initialized. What I'm having issues however is that the two commands described in the module "Can" and "CanSend" are both described as "Unknown" commands. 

I have enabled #use_CANSNIFFER and disabled #use_mcp2515, set the module GPIO's correctly, as I said, the connection's successful, but simply the commands aren't recognized. Not sure if there's anything else I need to add/modify?

**Comment by wolfing42** [comment]
Mcp2515.init_status = 1; is missing. for example:
bool Xsns87(uint32_t function) {
  bool result = false;

  if (FUNC_INIT == function) {
    MCP2515_Init();
    Mcp2515.init_status = 1;
  }
  else if (Mcp2515.init_status) {
...

**Reply by sfromis** [reply]
While setting the flag is indeed missing, it should instead be inserted at the end of the `MCP2515_Init` function, to only have it set if an error message was not printed instead of successful initialization. Good candidate for submitting a PR if it then works.

**Comment by arendst** [comment]
I'll fix it now.

**Comment by arendst** [comment]
My, my that took a while ;-)

Now fixed in v15.5.0.2 https://github.com/arendst/Tasmota/commit/3725d6eb47ee30763071022f60b40c3739796f7a
